src/pages/authentication/signup/signup.js

```diff
@@ -57,37 +58,36 @@ function SignUp(){
     return(
         <div>
             <Navbar />
-             <form action="">
+             <form  action="" onSubmit={signupHandler}>
         <div className="form">
             <div className="sign-up-form">
                 <h2>Create Account</h2>
                 <label for="firstName">Name
         <input type="text"
         placeholder="enter your name" 
-        name='name'
-    
-        onChange={formHandler}
-        required />
+        required
+        name='firstname' 
+        onChange={formHandler} 
+         />
        </label>
 
             
                 <label for="email-id">Email-id
         <input type="text" 
         placeholder="enter your mail address" 
+        required
         name="email-id" 
-        
         onChange={formHandler}
-
-        required  />
+         />
        </label>
                 <label for="password">Password
         <input 
         placeholder="enter your password" 
-        name="password" 
+        name="password"
+        required 
         type={passwordToggle.type}
-        onChange={formHandler}
-
-        required />
+        onChange={formHandler} 
+         />
         {passwordToggle.isEyeIcon ? (<i className="fa fa-eye" id="eye-icon"  onClick={togglePassword} aria-hidden="true"></i>) :(<i className="fa fa-eye-slash" id="eye-icon-slash" onClick={togglePassword} aria-hidden="true"></i>)}
```

Comment:
you can just toggle the class name of font awesome icon here instead of re-writing as the function is common as well